win_setupgui/src/gui.rs

```diff
@@ -109,6 +110,57 @@ impl ConfigGui {
         nwg::stop_thread_dispatch();
     }
 
+    fn test_connection(&self) {
+        let dsn_opts = DSNOpts::new(
```

Comment:
non-blocking: maybe this could be created after checking the inputs are valid, since it doesn't appear to be used in the match? not a big difference, but might be easier to read

Reply:
I decided to do some more cleanup! Thanks for the suggestion.